src/org/thoughtcrime/securesms/loki/redesign/activities/HomeActivity.kt

```diff
@@ -243,21 +246,44 @@ class HomeActivity : PassphraseRequiredActionBarActivity, ConversationClickListe
                             apiDatabase.removeLastDeletionServerID(publicChat.channel, publicChat.server)
                             ApplicationContext.getInstance(activity).lokiPublicChatAPI!!.leave(publicChat.channel, publicChat.server)
                         }
+
                         threadDatabase.deleteConversation(threadID)
+
                         MessageNotifier.updateNotification(activity)
                     }
                 }
             }
-            val handler = Handler()
-            handler.postDelayed(deleteThread, 5000)
-            val snackbar = Snackbar.make(activity.contentView, "Conversation Deleted", Snackbar.LENGTH_LONG)
-            snackbar.setAction("Undo") {
-                threadDatabase.unarchiveConversation(threadID)
-                handler.removeCallbacks(deleteThread)
-                animate(viewHolder, 0.0f)
+
+            val message = if (recipient.isGroupRecipient) R.string.activity_home_leave_group_title else R.string.activity_home_delete_conversation_title
+            val alertDialogBuilder = AlertDialog.Builder(activity)
+            alertDialogBuilder.setMessage(message)
+            alertDialogBuilder.setPositiveButton(R.string.yes) { _, _ ->
+                val isGroup = recipient.isGroupRecipient
+
+                // If we are deleting a group and it's active
+                // We need to send a leave message
+                if (isGroup && DatabaseFactory.getGroupDatabase(activity).isActive(recipient.address.toGroupString())) {
+                    if (!GroupUtil.leaveGroup(activity, recipient)) {
+                        Toast.makeText(activity, R.string.ConversationActivity_error_leaving_group, Toast.LENGTH_LONG).show()
+                        clearView(activity.recyclerView, viewHolder)
+                        return@setPositiveButton
+                    }
+                }
+
+                // Archive and forcefully delete the conversation in 10 seconds
+                threadDatabase.archiveConversation(threadID)
+                val handler = Handler()
+                handler.postDelayed(deleteThread, 10000)
+
+                // Notify the user
+                val snackbarText = if (isGroup) R.string.MessageRecord_left_group else R.string.activity_home_conversation_deleted
```

Comment:
Our convention is to use a toast instead of a snackbar if there's no action for the user to take